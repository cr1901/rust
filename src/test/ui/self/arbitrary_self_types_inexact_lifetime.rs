#![feature(arbitrary_self_types)]

#[derive(Debug)]
struct Foo;

impl Foo {
    fn a(self: &Box<Foo>, f: &Foo) -> &Foo { f } //~ ERROR E0106

    fn b(self: &Box<Foo>, f: &Foo) -> &Box<Foo> { self } //~ ERROR E0106

    fn c(this: &Box<Foo>, f: &Foo) -> &Foo { f } //~ ERROR E0106
}

fn main() {}
