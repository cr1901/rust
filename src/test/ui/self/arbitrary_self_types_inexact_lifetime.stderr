error[E0106]: missing lifetime specifier
  --> $DIR/arbitrary_self_types_inexact_lifetime.rs:7:39
   |
LL |     fn a(self: &Box<Foo>, f: &Foo) -> &Foo { f }
   |                                       ^ expected lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from `self` or `f`

error[E0106]: missing lifetime specifier
  --> $DIR/arbitrary_self_types_inexact_lifetime.rs:9:39
   |
LL |     fn b(self: &Box<Foo>, f: &Foo) -> &Box<Foo> { self }
   |                                       ^ expected lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from `self` or `f`

error[E0106]: missing lifetime specifier
  --> $DIR/arbitrary_self_types_inexact_lifetime.rs:11:39
   |
LL |     fn c(this: &Box<Foo>, f: &Foo) -> &Foo { f }
   |                                       ^ expected lifetime parameter
   |
   = help: this function's return type contains a borrowed value, but the signature does not say whether it is borrowed from `this` or `f`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0106`.
